packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1867,6 +1872,9 @@ class _TabBarViewState extends State<TabBarView> {
       _controller!.animation!.removeListener(_handleTabControllerAnimationTick);
     }
     _controller = null;
+    if (_pageControllerCreated) {
+      _pageController.dispose();
```

Comment:
should we set `_pageControllerCreated` back to `false` here?

Reply:
For my understanding `_pageControllerCreated` is needed because `dispose()` may be called before the controller is created in `didChangeDependencies`?

Reply:
I think this is fine, once it is disposed, it should not be called in didChangeDependencies again.

Reply:
That being said I don't think you need this property, didchangedependencies must be called before dispose is called. you should be able to assume _pageController is not null at this point.

Reply:
_pageController is not nullable and late
This means _pageControllerCreated should be set after the controller is created and when it is not disposed yet. 

Renamed to _pageControllerCreatedAndNotDisposed.

Reply:
If this is the case, it defeat the purpose of `late`, it can be null when it is actually used. I think either we assume the State will be used in the expected manner and directly call dispose without any check, or we convert it to nullable (you can have a getter to un-null it)

Reply:
didchangedependencies is invoked first time for non initialized value